import_export/resources.py

```diff
@@ -846,7 +846,8 @@ def import_data_inner(
         if collect_failed_rows:
             result.add_dataset_headers(dataset.headers)
 
-        for i, row in enumerate(dataset.dict, 1):
+        for i, data_row in enumerate(dataset, 1):
+            row = OrderedDict(zip(dataset.headers, data_row))
```

Comment:
Out of interest can you explain how this improves the memory performance?  Presumably the call to dataset.dict creates a copy in memory?

Reply:
Exactly!